### PR TITLE
Fix variants not getting removed

### LINKED ISSUE #163: Make sure variants get removed in database if they are removed in Control Panel
### Description
@ryangjchandler I wonder if we could do a similar thing that we did with the variant attributes where we do a check thing.

### PR DESCRIPTION
**Pull request by ryangjchandler**
Closes #163 .

This also refactors some of the code from the custom attributes changes to make it more streamlined and a bit less verbose, although it is still quite verbose.

### CODE REVIEW

File: src/Http/Controllers/Cp/ProductController.php [hunk 1]
```diff
@@ -179,17 +179,15 @@ public function update(ProductUpdateRequest $request, Product $product)
     {
         $this->authorize('update', $product);
 
-        $validation = $request->validated();
```
Comment (duncanmcclean) [inline src/Http/Controllers/Cp/ProductController.php:182]: Why did you remove the request validation?

File: src/Http/Controllers/Cp/ProductController.php [hunk 2]
```diff
@@ -204,49 +202,41 @@ public function update(ProductUpdateRequest $request, Product $product)
...
             });
 
-        $product->attributes
-            ->filter(function ($attribute) use ($requestAttributes) {
-                return ! $requestAttributes
-                    ->contains(function ($requestAttribute) use ($attribute) {
-                        return $attribute->key === $requestAttribute['key'] && $attribute->value === $requestAttribute['value'];
-                    });
+        $product->variants
+            ->filter(function ($variant) use ($requestVariants) {
```
Comment (duncanmcclean) [inline src/Http/Controllers/Cp/ProductController.php:237]: where is that getting `$requestVariants` from?

**Review by duncanmcclean — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/Cp/ProductController.php (hunk 1 above)
Comment (ryangjchandler) [reply src/Http/Controllers/Cp/ProductController.php:182]: There's no need to call that method. All it does is return the validated data, not actually validate the request, that gets done during the request lifecycle.

File: src/Http/Controllers/Cp/ProductController.php (hunk 2 above)
Comment (ryangjchandler) [reply src/Http/Controllers/Cp/ProductController.php:237]: > where is that getting `$requestVariants` from?

Line 189

File: src/Http/Controllers/Cp/ProductController.php (hunk 1 above)
Comment (duncanmcclean) [reply src/Http/Controllers/Cp/ProductController.php:182]: Ah ok, I always thought that was required 😃

**Comment by duncanmcclean** [comment]
I'll merge this pull request in a bit when I commit what I'm currently working on.

**Pull request merged**